x-pack/performance/journeys/aiops_log_rate_analysis.ts

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import * as estypes from '@elastic/elasticsearch/lib/api/typesWithBodyKey';
+
+import { Journey } from '@kbn/journeys';
+import { subj } from '@kbn/test-subj-selector';
+import type { DataView } from '@kbn/data-views-plugin/common';
+
+// We are importing from tests to tests here, not related to runtime code.
+// eslint-disable-next-line @kbn/imports/no_boundary_crossing
+import { frequentItemSetsLargeArraysSource } from '../../test/functional/apps/aiops/log_rate_analysis/test_data/__mocks__/frequent_item_sets_large_arrays';
+
+const indexName = 'large_arrays';
+let dataViewId: string | undefined;
+
+interface DocWithArray {
+  '@timestamp': number;
+  items: string[];
+}
+
+export const journey = new Journey({
+  beforeSteps: async ({ kbnUrl, log, auth, es, kibanaServer }) => {
+    // Create index with mapping
+    try {
+      await es.indices.create({
+        index: indexName,
+        mappings: {
+          properties: {
+            items: { type: 'keyword' },
+            '@timestamp': { type: 'date' },
+          },
+        },
+      });
+    } catch (e) {
+      log.error(`Error creating index '${indexName}' in beforeSteps() callback`);
+    }
```

Comment:
Since we run journey 2 times (warmup & actual test), but start Elastic search only one time:

maybe we first check if index exists and if not, create it? Same with dats ingestion?
It probably doesn't make sense to re-ingest the data for the 2nd time as well. WDYT?

p.s. I believe we do the same with esArchiver

Reply:
I'm not sure the run process is obvious, so I will share more details.

Originally we ran journey as a single run with ES and Kibana started before it. The metrics were to some degree inconsistent.

We thought we try to "warmup" Elasticsearch by running journey 2 times in a row. Results were improved and quite consistent, so we decided to start ES one time and start Kibana with journey as warmup first (no EBT collection) and later restart Kibana and repeat the same journey already collecting EBT.